core/src/test/java/org/apache/iceberg/TestGeospatialTable.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.hadoop.HadoopTableTestBase;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.EdgeInterpolationAlgorithm;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.junit.jupiter.api.Test;
+
+public class TestGeospatialTable extends HadoopTableTestBase {
```

Comment:
Why does this use `HadoopTableTestBase`? I would avoid using `HadoopCatalog` or considering something that works with `HadoopCatalog` to be working everywhere.

Reply:
Moved this test to `CatalogTests`.

Reply:
Yes, it validate catalog behavior. By moving it to CatalogTest we'll be able to test it against all catalog implementations. We found that HiveCatalog does not support geospatial types and fixed it in the previous commit. If you insist that we only test using `InMemoryCatalog`, I'll bring `TestGeospatialTable` back and test using `InMemoryCatalog` only.